core/src/main/java/org/radargun/stages/test/RunningTest.java

```diff
@@ -0,0 +1,290 @@
+package org.radargun.stages.test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+
+import org.radargun.logging.Log;
+import org.radargun.logging.LogFactory;
+import org.radargun.state.ServiceListenerAdapter;
+import org.radargun.stats.Statistics;
+import org.radargun.utils.TimeService;
+import org.radargun.utils.Utils;
+
+/**
+ * State of test (threads, synchronization objects etc..) that is created by {@link TestSetupStage}
+ * and stored in {@link org.radargun.state.SlaveState} for the duration of the test, which usually
+ * spans several stages (on or more pairs of {@link TestSetupStage} and {@link TestStage} and possibly
+ * {@link FinishTestStage}).
+ *
+ * @author Radim Vansa &lt;rvansa@redhat.com&gt;
+ */
+public class RunningTest extends ServiceListenerAdapter {
+   private final Log log = LogFactory.getLog(RunningTest.class);
+   private final ConcurrentLinkedQueue<Statistics> statisticsQueue = new ConcurrentLinkedQueue<>();
+   private final ArrayList<Stressor> stressors = new ArrayList<>();
+   private final ThreadLocal<Conversation> lastConversation = new ThreadLocal<>();
+   private final AtomicLong lastCreated = new AtomicLong(Long.MIN_VALUE);
+   private final ConcurrentMap<Conversation, AtomicInteger> actuallyExecuting = new ConcurrentHashMap<>();
+   private final ArrayList<Runnable> stopListeners = new ArrayList<>();
+
+   private Statistics statisticsPrototype;
+   private AtomicInteger waitingThreads = new AtomicInteger();
+   private AtomicInteger threadCounter = new AtomicInteger();
+   private AtomicInteger statisticsCounter = new AtomicInteger();
+
+   private volatile boolean steadyState = false;
```

Comment:
The class is quite complex, it would be worth to at least add comments to describe the fields (or find better names which will avoid the need of having those comments).